### DISCUSSION TITLE
What Markdown engine is used in Slidev?

### DISCUSSION THREAD

**Discussion by spences10** [slidevjs/slidev#435] [Q&A] [answered]
I have a presentation which doesn't recognise svelte code fences:

so, something like this:

![image](https://user-images.githubusercontent.com/234708/145633138-df47ad93-15ef-4cdf-bf06-ea359118d78d.png)

Should give a result like this:

![image](https://user-images.githubusercontent.com/234708/145633009-3b557352-b566-4cdf-b1db-abc7e97c06f3.png)

Instead it looks like this:

![image](https://user-images.githubusercontent.com/234708/145634857-e2c31f7a-b852-467c-b590-d9c008b79166.png)

But there doesn't seem to be any support for it

Is there a way to specify the markdown languages?

**Comment by antfu** [comment, marked as answer]
https://sli.dev/custom/highlighters.html

**Reply by spences10** [reply]
Hi @antfu, that's for highlighting, where's the language support?

So as an example in VS Code there's an extension for Svelte support for highlighting, I have noticed in my blog (which uses MDSveX) that highlighting for Svelte is supported, so I'm asking if it's something that needs to be enabled somewhere?

**Reply by antfu** [reply]
Isn't your request is the highlighting for Svelte? Both Prism and Shiki have syntax builtin, which they many have API to provide custom syntax.

**Reply by spences10** [reply]
No, my request isn't about highlighting, I haven't mentioned syntax highlighting at any point. 😊

I'm asking if there is a way to have the Svelte language recognised by Slidev.

Currently, regardless of highlighter used, Svelte syntax isn't recognised.

**Reply by spences10** [reply]
So @pngwn has this: https://github.com/pngwn/prism-svelte

I'm going to try figure out where this can be implemented in Slidev

**Reply by pngwn** [reply]
Hey. I think the question is how would you pass a custom grammar to Prism? Prism doesn't handle svelte syntax out of the box. The way Prism work is pretty odd (by modern standards) so just importing it at the right time is probably enough. 

Shiki does support svelte out of the box, iirc, so just switching to shiki would probably work.

**Reply by spences10** [reply]
Sweet! That does it @pngwn 🙏

I changed the highlighter over to Shiki! 

I've imported my own json file for Nigh Owl!

Thanks for Slidev @antfu 🙏
